### PR TITLE
PLT-2380 Improvements to webpack build system + watch mode

### PR DESCRIPTION
**Pull request by crspeller**
- Enabled watch mode.
- Upgraded to webpack master to fix watch mode bug
- Created seperate buids for development and release

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -49,11 +49,11 @@
     "node-sass": "3.4.2",
     "sass-loader": "3.2.0",
     "style-loader": "0.13.0",
-    "webpack": "2.1.0-beta.4"
+    "webpack": "webpack/webpack#master"
```
Comment (hmhealey) [inline webapp/package.json]: We might want this set to a specific commit number just in case something breaks with webpack again. 0/5 on that though

**Comment by hmhealey** [comment]
+1 other than my question

File: webapp/package.json (hunk 1 above)
Comment (crspeller) [inline webapp/package.json]: Good point. Meant to change that to lock to a commit. Will do.

**Comment by jwilander** [comment]
+1

File: webapp/package.json (hunk 1 above)
Comment (crspeller) [inline webapp/package.json]: Did not work. Reverted.

**Pull request merged**